### PR TITLE
chore(cf): add integration tests

### PR DESCRIPTION
**Pull request by ssddOnTop**
Original issue: #750 

Steps to run: `npm install && npm test` in cloudflare dir

### CODE REVIEW

File: cloudflare/src/http.rs [hunk 1]
```diff
@@ -78,6 +78,7 @@ pub fn to_method(method: worker::Method) -> anyhow::Result<hyper::Method> {
 
 pub async fn to_request(mut req: worker::Request) -> anyhow::Result<hyper::Request<hyper::Body>> {
   let body = req.text().await.map_err(to_anyhow)?;
+  log::info!("{:?}", body);
```
Comment (tusharmath) [inline cloudflare/src/http.rs:81]:
```suggestion
```

File: cloudflare/Cargo.toml [hunk 2]
```diff
@@ -24,6 +24,11 @@ serde_json = "1.0.111"
 serde_qs = "0.12.0"
 console_error_panic_hook = "0.1.7"
 
+[dev-dependencies]
+wasm-bindgen-test = "0.3.28" # using lower version compatible with worker crate
+gloo-utils = {version = "0.2.0",features = ["serde"]}
+js-sys = "0.3.63"
+
```
Comment (tusharmath) [inline cloudflare/Cargo.toml:31]:
```suggestion
wasm-bindgen-test = "0.3.28" # using lower version compatible with worker crate
```

**Review by tusharmath — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -43,8 +43,21 @@ jobs:
...
-      - name: Build WASM
-        run: cargo build --lib --release --target wasm32-unknown-unknown
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.11.0"
+
+      - name: Compile CF
+        run: cargo install -q worker-build && worker-build
+        working-directory: ./cloudflare
```
Comment (meskill) [inline .github/workflows/ci.yml:53]: we can specify step defaults on job level https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iddefaults

File: cloudflare/tests/cf_tests.spec.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+describe("fetch", () => {
+  test("loadfiles", async () => {
+    let placeholder = (await readFile("../examples/jsonplaceholder.graphql")).toString()
+    let placeholder_batch = (await readFile("../examples/jsonplaceholder_batch.graphql")).toString()
+
+    let bucket = await mf.getR2Bucket("MY_R2")
+    // @ts-ignore
+    await bucket.put("examples/jsonplaceholder.graphql", placeholder)
+    // @ts-ignore
```
Comment (meskill) [inline cloudflare/tests/cf_tests.spec.ts:13]: install additional type packages to fetch proper types and remove ignore

```
npm i --dev @cloudflare/workers-types
```

File: cloudflare/tests/cf_tests.spec.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    })
+    let body = await resp.text()
+    expect(body.includes("<title>Tailcall - GraphQL IDE</title>")).toBe(true)
+    expect(resp.status).toBe(200)
+  })
+
+  test("sample_resp", async () => {
+    let resp = await mf.dispatchFetch("https://fake.host/graphql?config=examples/jsonplaceholder.graphql", {
+      method: "POST",
+      body: '{"query":"{\\n  user(id: 1) {\\n    id\\n  }\\n}\\n"}',
```
Comment (meskill) [inline cloudflare/tests/cf_tests.spec.ts:28]: the `\n` are not necessary here. Could be easier to read if we remove it. Also we can use different way to specify it like template strings or JSON.stringify for big requests

File: cloudflare/tests/cf_tests.spec.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    expect(resp.status).toBe(200)
+  })
+
+  test("test_batching", async () => {
+    let resp = await mf.dispatchFetch("https://fake.host/graphql?config=examples/jsonplaceholder_batch.graphql", {
+      method: "POST",
+      body: '{"query":"{\\n  posts {\\n    id\\n  }\\n}\\n"}',
+    })
+    let body = await resp.text()
+    expect(body).toBe('{"data":{"posts":[{"id":1}]}}')
```
Comment (meskill) [inline cloudflare/tests/cf_tests.spec.ts:41]: could use `resp.json` to convert to js object and compare with object to not be forced on whitespaces etc.

**Pull request merged**